src/gsi/read_l2bufr_mod.f90

```diff
@@ -302,12 +303,6 @@ subroutine radar_bufr_read_all(npe,mype)
 !   Open bufr file with openbf to initialize bufr table, etc in bufrlib
     inbufr=10
     open(inbufr,file=infile,form='unformatted')
-    read(inbufr,iostat=iret)subset
-    if(iret/=0) then
```

Comment:
After this fix,  the subset will not  be initialized. However, in the new line 338 block,  subset will be used.  Will this also be taken care of , like adding a  new logical branch?

Reply:
Here, read(inbufr,iostat=iret)subset is not initialize "subset", it is to test if the file is readable. "subset" is declared as "character (8)", the first character is not necessary character(8). This caused the failure of iostat test. In line 338, "ireadmg" in bufr lib can handle this.

Reply:
Got it , thanks. So, in the line 338,  do while(ireadmg(inbufr,subset,idate)>=0),  the subset  is to be read out rather than to be used as an intent-in argument.